### PR TITLE
Refactor IRS incomes and add Medicaid income

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
In order to get the Medicaid MAGI right (and to make the IRS section of the codebase more maintainable), I'm aiming to simplify and refactor the tax inputs, tying to legislation where possible. This PR will add the variables for IRS gross income, AGI (already exists, but refactored a bit) and Medicaid-specific MAGI and SS-specific MAGI.

cc @MaxGhenis, @rickecon

### CODE REVIEW

File: openfisca_us/variables/irs/income/gross/tax_unit_farm_income.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Tax unit farm income"
+    unit = USD
+    documentation = "Tax unit farm income (excludes dependents)."
+    definition_period = YEAR
+
+    formula = sum_among_non_dependents("farm_income")
+
+tc_e02100 = taxcalc_read_only_variable("tc_e02100", tax_unit_farm_income)
```
Comment (MaxGhenis) [inline openfisca_us/variables/irs/income/gross/tax_unit_farm_income.py:14]: I'd slightly prefer taxcalc to tc as a prefix for clarity (same with taxsim)

**Review by MaxGhenis — APPROVED** [review]

**Comment by nikhilwoodruff** [comment]
@MaxGhenis the implementation is now mainly done (some things might have changed since your review), this just needs revisiting the TAXSIM integration tests for taxable social security, some of which are now failing. I'll post in this thread where it's unclear to me why.

**Comment by nikhilwoodruff** [comment]
Fixed the bug - SS-related MAGI didn't include UI because I misread [26 U.S. Code § 86(b)(2)(A)](https://www.law.cornell.edu/uscode/text/26/86#b_2_A) - I thought it said to disregard section 85 ("include UI in gross income"), but it actually just says to disregard 85(c) ("Some UI is not taxable if X"). Passing now.

**Comment by nikhilwoodruff** [comment]
Including medicaid income in this PR because the failures are only on GH, tests passing locally.

File: openfisca_us/variables/contrib/taxsim/v11.py [hunk 2]
```diff
@@ -9,4 +9,4 @@ class v11(Variable):
     definition_period = YEAR
 
     def formula(tax_unit, period, parameters):
-        return tax_unit("taxable_ui", period)
+        return tax_unit("tax_unit_taxable_unemployment_insurance", period)
```
Comment (MaxGhenis) [inline openfisca_us/variables/contrib/taxsim/v11.py:12]: If we're renaming it, should we go with `unemployment compensation`, the term in [IRS documents](https://www.irs.gov/forms-pubs/exclusion-of-up-to-10200-of-unemployment-compensation-for-tax-year-2020-only) and the US Code?

File: openfisca_us/variables/gov/irs/income/sources/social_security/tax_exempt_social_security.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from openfisca_us.model_api import *
+
+
+class tax_exempt_social_security(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Tax-exempt social security"
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/irs/income/sources/social_security/tax_exempt_social_security.py:7]: Looks like the IRS is inconsistent about capitalization, but I typically see it capitalized.
```suggestion
    label = "Tax-exempt Social Security"
```

File: openfisca_us/variables/gov/irs/income/sources/social_security/taxable_social_security.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        # The taxable amount of social security is decided at the tax unit level, but
+        # gross income (which contains taxable SS) is person-level. Therefore, we include the
+        # taxable SS in gross income by assigning it to the head of the tax unit: this will be
+        # not affect overall tax liability.
+
+        is_tax_unit_head = person("is_tax_unit_head", period)
+        tax_unit_taxable_ui = person.tax_unit(
+            "tax_unit_taxable_social_security", period
+        )
+        return where(is_tax_unit_head, tax_unit_taxable_ui, 0)
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/irs/income/sources/social_security/taxable_social_security.py:21]: Our more typical pattern
```suggestion
        return is_tax_unit_head * tax_unit_taxable_ui
```

File: openfisca_us/variables/gov/irs/income/sources/unemployment_insurance/taxable_ui_agi.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+from openfisca_us.model_api import *
+
+
+class taxable_ui_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Adjusted gross income (SS)"
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/irs/income/sources/unemployment_insurance/taxable_ui_agi.py:7]:
```suggestion
    label = "Taxable unemployment insurance for SS adjusted gross income"
```

File: openfisca_us/variables/household/income/person/pensions/pension_income.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+class pension_income(Variable):
+    value_type = float
+    entity = Person
+    label = "Pension income"
+    unit = USD
+    documentation = "Income from pensions, annuitities, life insurance or endowment contracts."
+    definition_period = YEAR
+
+    formula = sum_of_variables(["ira_income"])  # TODO: add other pensions
```
Comment (MaxGhenis) [inline openfisca_us/variables/household/income/person/pensions/pension_income.py:12]: Is this todo for this PR, or for an issue?

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**